### PR TITLE
Add M5Stack-FIRE board

### PR DESCRIPTION
**Pull request by 0x1abin**
Add M5Stack-FIRE board 16MB FLASH, 4MB PSRAM.

### CODE REVIEW

File: variants/m5stack_fire/pins_arduino.h [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+static const uint8_t G36 = 36;
+static const uint8_t G25 = 25;
+static const uint8_t G26 = 26;
+static const uint8_t G1 = 1;
+static const uint8_t G17 = 17;
+static const uint8_t G22 = 22;
+static const uint8_t G5 = 5;
+static const uint8_t G13 = 13;
+static const uint8_t G0 = 0;
+static const uint8_t G34 = 34;
```
Comment (me-no-dev) [inline variants/m5stack_fire/pins_arduino.h]: why are those?

File: variants/m5stack_fire/pins_arduino.h [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+#define analogInputToDigitalPin(p)  (((p)<20)?(esp32_adc2gpio[(p)]):-1)
+#define digitalPinToInterrupt(p)    (((p)<40)?(p):-1)
+#define digitalPinHasPWM(p)         (p < 34)
+
+static const uint8_t TX = 1;
+static const uint8_t RX = 3;
+
+static const uint8_t TXD2 = 17;
+static const uint8_t RXD2 = 16;
```
Comment (me-no-dev) [inline variants/m5stack_fire/pins_arduino.h]: 16 and 17 are connected to PSRAM

File: boards.txt [hunk 3]
```diff
@@ -1553,6 +1553,93 @@ m5stack-core-esp32.menu.DebugLevel.verbose.build.code_debug=5
...
+m5stack-fire.menu.FlashMode.qout.build.flash_mode=dout
+m5stack-fire.menu.FlashMode.qout.build.boot=qout
+m5stack-fire.menu.FlashMode.dout=DOUT
+m5stack-fire.menu.FlashMode.dout.build.flash_mode=dout
+m5stack-fire.menu.FlashMode.dout.build.boot=dout
+
+m5stack-fire.menu.FlashFreq.80=80MHz
+m5stack-fire.menu.FlashFreq.80.build.flash_freq=80m
+m5stack-fire.menu.FlashFreq.40=40MHz
+m5stack-fire.menu.FlashFreq.40.build.flash_freq=40m
```
Comment (me-no-dev) [inline boards.txt]: find what works on the board and let's remove this option.

File: boards.txt [hunk 4]
```diff
@@ -1553,6 +1553,93 @@ m5stack-core-esp32.menu.DebugLevel.verbose.build.code_debug=5
...
+m5stack-fire.menu.FlashMode.qio.build.boot=qio
+m5stack-fire.menu.FlashMode.dio=DIO
+m5stack-fire.menu.FlashMode.dio.build.flash_mode=dio
+m5stack-fire.menu.FlashMode.dio.build.boot=dio
+m5stack-fire.menu.FlashMode.qout=QOUT
+m5stack-fire.menu.FlashMode.qout.build.flash_mode=dout
+m5stack-fire.menu.FlashMode.qout.build.boot=qout
+m5stack-fire.menu.FlashMode.dout=DOUT
+m5stack-fire.menu.FlashMode.dout.build.flash_mode=dout
+m5stack-fire.menu.FlashMode.dout.build.boot=dout
```
Comment (me-no-dev) [inline boards.txt]: find what works on the board and let's remove this option. DIO or QIO

File: boards.txt [hunk 5]
```diff
@@ -1553,6 +1553,93 @@ m5stack-core-esp32.menu.DebugLevel.verbose.build.code_debug=5
...
+m5stack-fire.menu.PartitionScheme.default=Default
+m5stack-fire.menu.PartitionScheme.default.build.partitions=default
+m5stack-fire.menu.PartitionScheme.minimal=Minimal (2MB FLASH)
+m5stack-fire.menu.PartitionScheme.minimal.build.partitions=minimal
+m5stack-fire.menu.PartitionScheme.no_ota=No OTA (Large APP)
+m5stack-fire.menu.PartitionScheme.no_ota.build.partitions=no_ota
+m5stack-fire.menu.PartitionScheme.no_ota.upload.maximum_size=2097152
+m5stack-fire.menu.PartitionScheme.min_spiffs=Minimal SPIFFS (Large APPS with OTA)
+m5stack-fire.menu.PartitionScheme.min_spiffs.build.partitions=min_spiffs
+m5stack-fire.menu.PartitionScheme.min_spiffs.upload.maximum_size=1966080
```
Comment (me-no-dev) [inline boards.txt]: those partitions make no sense. let's discuss this and add new options :)

File: boards.txt [hunk 6]
```diff
@@ -1553,6 +1553,93 @@ m5stack-core-esp32.menu.DebugLevel.verbose.build.code_debug=5
...
+m5stack-fire.build.flash_freq=40m
+m5stack-fire.build.flash_mode=dio
+m5stack-fire.build.boot=dio
+m5stack-fire.build.partitions=default
+m5stack-fire.build.defines=
+
+m5stack-fire.menu.PSRAM.disabled=Disabled
+m5stack-fire.menu.PSRAM.disabled.build.defines=
+m5stack-fire.menu.PSRAM.enabled=Enabled
+m5stack-fire.menu.PSRAM.enabled.build.defines=-DBOARD_HAS_PSRAM -mfix-esp32-psram-cache-issue
```
Comment (me-no-dev) [inline boards.txt]: Put enabled on top so it's the default option and why not enable it by default by adding the defines to the main section?

File: variants/m5stack_fire/pins_arduino.h (hunk 2 above)
Comment (0x1abin) [reply variants/m5stack_fire/pins_arduino.h]: Yes, it should remove.

File: boards.txt (hunk 5 above)
Comment (0x1abin) [reply boards.txt]: Yes, there is no partition for 16MB Flash, should add new options.

File: variants/m5stack_fire/pins_arduino.h (hunk 2 above)
Comment (me-no-dev) [reply variants/m5stack_fire/pins_arduino.h]: I think you forgot those lines :)

**Comment by me-no-dev** [comment]
@0x1abin I would like to merge this. Can you please remove pins 16 and 17 from the variant header?

**Pull request merged**